File: backend/myPizza/src/main/java/backendapp/myPizza/controllers/ProfileController.java
package backendapp.myPizza.controllers;

import backendapp.myPizza.Models.entities.Address;
import backendapp.myPizza.Models.entities.Order;
import backendapp.myPizza.Models.entities.User;
import backendapp.myPizza.Models.reqDTO.ChangePasswordDTO;
import backendapp.myPizza.Models.reqDTO.UserPutDTO;
import backendapp.myPizza.Models.resDTO.AdminUserIdRes;
import backendapp.myPizza.Models.resDTO.AuthoritiesRes;
import backendapp.myPizza.Models.resDTO.ConfirmRes;
import backendapp.myPizza.Models.resDTO.IsLoggedInRes;
import backendapp.myPizza.SocketIO.entities.Message;
import backendapp.myPizza.SocketIO.services.MessageService;
import backendapp.myPizza.exceptions.BadRequestException;
import backendapp.myPizza.exceptions.NotFoundException;
import backendapp.myPizza.exceptions.UnauthorizedException;
import backendapp.myPizza.services.ProfileService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.UUID;

@RestController
@RequestMapping("/api/user-profile")
public class ProfileController {

    @Autowired
    private ProfileService profileSvc;

    @Autowired
    private MessageService messageSvc;

    @GetMapping("")
    public User get() throws UnauthorizedException {
        return profileSvc.get();
    }

    @PutMapping("")
    public User put(@RequestBody UserPutDTO userPutDTO) throws UnauthorizedException, BadRequestException {
        return profileSvc.put(userPutDTO);
    }

    @DeleteMapping("")
    public ConfirmRes delete() throws UnauthorizedException, BadRequestException {
        return profileSvc.delete();
    }

    @GetMapping("/orders")
    public List<Order> getOrders() throws UnauthorizedException {
        return profileSvc.getOrders();
    }

    @GetMapping("/addresses")
    public List<Address> getAddresses() throws UnauthorizedException {
        return profileSvc.getAddresses();
    }

    @PatchMapping("/change-password")
    public ConfirmRes changePassword(@RequestBody ChangePasswordDTO changePasswordDTO) throws UnauthorizedException, BadRequestException {
        return profileSvc.changePassword(changePasswordDTO.oldPassword(), changePasswordDTO.newPassword());
    }

    @GetMapping("/is-logged-in")
    public IsLoggedInRes isLoggedIng() {
        return new IsLoggedInRes(true);
    }

    @GetMapping("get-authorities")
    public AuthoritiesRes getAuthorities() throws UnauthorizedException {
        return new AuthoritiesRes(profileSvc.getAuthorities());
    }

    @GetMapping("get-admin-userid")
    public AdminUserIdRes getAdminUserId() throws NotFoundException {
        return profileSvc.getAdminUserId();
    }

    // MESSAGES

    @GetMapping("/message/{id}/set-read")
    public Message setMessageRead(@PathVariable UUID id) throws NotFoundException {
        return messageSvc.getMessageById(id);
    }

}

File: backend_nodejs_nest_fastify/my-pizza/src/nest_modules/auth-user/controllers/profile.controller.ts
import { Controller, Get, Req } from '@nestjs/common';
import { FastifyRequest } from 'fastify';
import { UserRes } from '../interfaces/user-res.interface';
import { ProfileService } from '../services/profile.service';
import { IsLoggedInRes } from '../interfaces/is-logged-in-res.interface';
import { AdminUserIdRes } from '../interfaces/admin-user-id-res-interface';
import { AuthoritiesRes } from '../interfaces/authorities-res.interface';

@Controller('api/user-profile')
export class ProfileController {

    constructor(private profileSvc: ProfileService) { }

    @Get()
    public async getProfile(@Req() req: FastifyRequest): Promise<UserRes> {
        return this.profileSvc.getProfile(req)
    }

    @Get('/is-logged-in')
    public isLoggedIn(): IsLoggedInRes {
        return {
            loggedIn: true
        }
    }

    @Get('/get-admin-userid')
    public async getAdminUserId(): Promise<AdminUserIdRes> {
        return await this.profileSvc.getAdminUserId()
    }

    @Get('/get-authorities')
    public async getAuthorities(@Req() res: FastifyRequest): Promise<AuthoritiesRes> {
        return await this.profileSvc.getAuthorities(res)
    }



    



}
